pkg/operatorsource/phase_transitioner.go

```diff
@@ -0,0 +1,71 @@
+package operatorsource
+
+import (
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// NewPhaseTransitionerWithClock returns a new PhaseTransitioner with the given clock.
+// This is useful when you want to inject your own implementation of clock.Clock
```

Comment:
Why do we need to inject our own `clock` implementation?

Reply:
We don't have our own implementationof clock. `NewTransitionerWithClock` allows a user to inject a `clock` which is helpful for unit testing. Please see `transitioner_test.go`.

Reply:
Then please update the documentation to say this done for unit testing.